implementations/rust/ockam/ockam_command/src/node/create.rs

```diff
@@ -347,32 +346,34 @@ async fn run_foreground_node(
         }
     }
 
-    if cmd.exit_on_eof {
-        stop_node_on_eof(&mut ctx, &opts, &node_name).await?;
-    }
-
-    Ok(())
-}
+    // Create a channel for communicating back to the main thread
+    let (tx, mut rx) = tokio::sync::mpsc::channel(2);
+
+    // Register a handler for the Ctrl+C signal
+    let tx_clone = tx.clone();
+    ctrlc::set_handler(move || {
+        tx_clone.blocking_send(1).unwrap();
+    })
+    .expect("Error setting Ctrl+C handler");
+
+    // Spawn a thread to monitor STDIN for EOF
+    let tx_clone = tx.clone();
+    std::thread::spawn(move || {
+        let mut buf = String::new();
+        match io::stdin().read_to_string(&mut buf) {
+            Ok(_) => tx_clone.blocking_send(1).unwrap(),
+            Err(e) => eprintln!("Error reading STDIN: {}", e),
+        };
+    });
```

Comment:
If it does what it has to do, pressing `ctrl+d` should trigger the EOF handler.

Reply:
I can change that to your version, but I thought since `read_to_string()` says that it 

```
Read(s) all bytes until EOF in this source, appending them to buf.

If successful, this function returns the number of bytes which were read and appended to buf.
```

I figured that it would jut continue to read `STDIN` until it receives `EOF` from somewhere and not exit on any line?

Reply:
mm good point. To be honest I'm not sure of the differences between the two methods for our use-case. You can keep your version and try if it works as expected. If it reacts to `ctrl+d` then we are good.

Reply:
I think mine was working too, but I changed it to yours because  the function name `read_to_end()` is more closely related to what we actually want to do :smile: